### ISSUE TITLE
Define the skeleton structure

### ISSUE THREAD

**Issue by nathanmarcos** [socialbase/crilex#1]

**Comment by romeumattos** [comment]
Sugiro Factory também

**Comment by darkSasori** [comment]
Repository não seria responsabilidade do ORM?

**Comment by romeumattos** [comment]
O que acham dessa [Sliex-api-skel/](https://github.com/mrprompt/silex-api-skel/tree/master/src/User/src)? @nathanmarcos @darkSasori

**Comment by nathanmarcos** [comment]
Concordo com você @darkSasori, porém, creio que para atingir nosso objetivo principal de "gerar um crud com o menor esforço possível", seria interessante já ter a classe de repositório com os métodos necessários para o funcionamento do crud e por hora assumimos como dependência do nosso gerador o uso do doctrine.

**Comment by nathanmarcos** [comment]
Achei bem bacana @romeumattos, mas a estrutura por si só não atende nossas necessidades, assim como algumas das estratégias utilizadas, mas já é uma boa base para iniciarmos, o que acham?

**Comment by nathanmarcos** [comment]
Estava pensando em utilizarmos [FormFactory](http://silex.sensiolabs.org/doc/providers/form.html) para efetuar o bind entre request e entity assim como efetuar a validação de formulário e para substituir os DTO's utilizarmos [Serializer](http://silex.sensiolabs.org/doc/providers/serializer.html).
